### PR TITLE
Labels on top

### PR DESCRIPTION
**Pull request by alonsogarciapablo**

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4699/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4702/))

File: app/models/map.rb [hunk 1]
```diff
@@ -267,9 +267,9 @@ def admits_more_torque_layers?
     torque_layers.length >= 1 && is_table_visualization? ? false : true
   end
 
+  # TODO: We need to allow more than one `tiled` layer now
```
Comment (alonsogarciapablo) [inline app/models/map.rb]: This will be fixed in #4256. @Kartones, you can create PR against this branch (`labels-on-top`).

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4705/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4706/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4718/))

File: lib/assets/javascripts/cartodb/models/map.js [hunk 2]
```diff
@@ -466,32 +472,6 @@ cdb.admin.Layers = cdb.geo.Layers.extend({
     saveLayer(layers.at(count));
   },
 
-  /**
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/models/map.js]: `layers` have an `order` attribute, [which is updated every time a new layer is added to the collection](https://github.com/CartoDB/cartodb.js/pull/555/files#diff-7cc9c3bb363716173690e1aaaae3786aR235). Unless I'm missing something, I think we can get rid of this custom `reset` and assume [layers are ordered](https://github.com/CartoDB/cartodb.js/blob/develop/src/geo/map.js#L222-L224).

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4720/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4722/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4724/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4725/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4753/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4755/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4757/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4765/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4770/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4784/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4790/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4799/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4813/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4820/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4834/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4850/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4864/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4894/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4906/))

**Comment by alonsogarciapablo** [comment]
This should be now ready to be merged! \o/

@xavijam would you mind taking a look at it? Thanks, thanks and thanks!

File: app/controllers/carto/api/users_controller.rb [hunk 3]
```diff
@@ -69,7 +69,6 @@ def render_auth_users_data(user, referrer, subdomain, referrer_organization_user
           email: user.email,
           organization: Carto::Api::OrganizationPresenter.new(user.organization).to_poro,
           base_url: user.public_url,
-          feature_flags: user.feature_flag_names
```
Comment (xavijam) [inline app/controllers/carto/api/users_controller.rb]: oh! why?

Comment (xavijam) [inline app/controllers/carto/api/users_controller.rb]: ping @kartones

File: lib/assets/javascripts/cartodb/models/cartodb_layer.js [hunk 4]
```diff
@@ -502,13 +502,14 @@ cdb.admin.CartoDBLayer = cdb.geo.CartoDBLayer.extend({
...
   moveToFront: function(opts) {
-    this.save({
-      'order': this.collection.last().get('order') + 1
-    }, opts);
-  },
-
-  //save: Backbone.saveAbort
+    var layers = this.collection;
 
+    // Removes and readds the layer at the top of layers
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/models/cartodb_layer.js]: typo :)

File: lib/assets/javascripts/cartodb/public_map/public_map_view.js [hunk 5]
```diff
@@ -45,74 +44,13 @@ module.exports = cdb.core.View.extend({
...
-    }
-  },
-
   _createVis: function() {
     var loadingTime  = cartodb.core.Profiler.metric('cartodb-js.embed.time_full_loaded').start();
     var visReadyTime = cartodb.core.Profiler.metric('cartodb-js.embed.time_vis_loaded').start();
     var self = this;
 
-    cartodb.createVis('map', this.options.vizdata, this.options.map_options, function(vis, layers) {
+    cartodb.createVis('map', this.options.vizdata, this.options.map_options, function(vis) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/public_map/public_map_view.js]: Ok, so we do not need labels on top specific code anymore, yay!

File: app/controllers/carto/api/users_controller.rb (hunk 3 above)
Comment (alonsogarciapablo) [inline app/controllers/carto/api/users_controller.rb]: Part of this PR is removing [this temporary hack](https://github.com/CartoDB/cartodb/commit/921e357647da7645d54259ecc70df44397c96ef1) and this line was added then.

Comment (Kartones) [inline app/controllers/carto/api/users_controller.rb]: No idea :/

Comment (xavijam) [inline app/controllers/carto/api/users_controller.rb]: Have you checked everything is working properly? If so :+1:.

**Comment by xavijam** [comment]
And, have you checked google basemaps (under google_maps flag) are working properly?

**Comment by alonsogarciapablo** [comment]
Yes! Google basemaps work :+1:

**Comment by xavijam** [comment]
In that case :uk:

**Comment by alonsogarciapablo** [comment]
Thanks @xavijam!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4933/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4934/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4935/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4936/))

**Pull request merged**